### ISSUE TITLE
Player turn:  allow operator replacement

### ISSUE THREAD

**Issue by donalgrant** [donalgrant/p6-equations#5]
Currently, the equation replacement is only for a single missing digit.  Extend _find_replacement_ method to operator replacement.  From the comments in the code:

```
# can we construct a missing operator with an equation representing
#     is inverse?  (can't do it for '-' and '/')
#   x+w --> x-(y-z)   where w = z-y
#   x*w --> x/(y/z)   where w = z/y
#   x^w --> (y/z)@x   where w = z/y
#   w@x --> (x^(y/z)) where w = z/y
```

**Comment by donalgrant** [comment]
implementation completed by commit b74122d; still need to complete the Board_Solver.t tests of the find_replacement sub, which has been moved from Player to Board_Solver

**Comment by donalgrant** [comment]
While lots of progress has been made on this, when a "sub-goal" involves something like 1/3, the replacement isn't successful, probably because the goal is something like 0.333333, which doesn't match exactly.  Either need to have goal check be Numeric-approximate, or have the goal actually be a Rat rather than Numeric.  (Or some other approach.)

**Comment by donalgrant** [comment]
fractional goal issue addressed; added tests for every digit and op replacement case.  Resolved with commit e7b9e51
